src/coreclr/jit/morph.cpp

```diff
@@ -12759,6 +12770,8 @@ GenTree* Compiler::fgMorphTree(GenTree* tree, MorphAddrContext* mac)
             break;
 
         case GT_SELECT:
+            assert(opts.OptimizationEnabled());
+
```

Comment:
Why this assert? `GT_SELECT` should be fully supported in the frontend, I see no reason to make it illegal to create this node in tier 0.

Reply:
> Why this assert? `GT_SELECT` should be fully supported in the frontend, I see no reason to make it illegal to create this node in tier 0.

the assert assumes that if-conversion phase is not run in Tier0, there is no value in this assert probably, just inserted to validate my assumption

Reply:
because this `case GT_SELECT` in morph uses `gtFoldExpr` and I guarded them all to avoid JTRUE(0/1) case

Reply:
> the assert assumes that if-conversion phase is not run in Tier0

It may be the case we only create GT_SELECT in if-conversion today, but the node should be usable in other cases in the future too (e.g. #81252). So I don't think we should assert that this node is illegal to have, even in tier 0. 

> because this case GT_SELECT in morph uses gtFoldExpr and I guarded them all to avoid JTRUE(0/1) case

I see, I guess I'd suggest we just guard the `gtFoldExpr` call like in the other cases then.